Orange/widgets/unsupervised/owsom.py

```diff
@@ -774,6 +774,7 @@ def _assign_instances(self, weights, ssum_weights):
         for i, (x, y) in enumerate(assignments):
             members[(x, y)].append(i)
         members.pop(None, None)
+        self.coordinates = np.array([[x, y] for i, (x, y) in enumerate(assignments)], dtype=int)
```

Comment:
1. What am I overlooking? Doesn't this just copy assignments, as with `self.coordinates = assignments.copy()`?
2. If you do this here, it will record coordinates on each step of optimization. I think this is not what you want, because coordinates are only needed for output.
3. `self.coordinates` stores the same information that is already stored in `self.member_data` and `self.cells` (combined). In my experience, it is better to have a *single source of truth*; if you store data in different places it may be too difficult to keep it synchronized and all too easy to forget to remove it. Case in point, method `clear` should include `self.coordinates = None`.

I suggest removing this line and reconstruct coordinates in `update_output`.

More so because reconstructing coordinates from `self.member_data` and `self.cells` in one sweep (without loops) could be a great exercise in `numpy` indexing, I think. :)